### PR TITLE
Crecto replacement.

### PR DESCRIPTION
**Pull request by Lukc**
Replaced by a `passwd` and `group` files reader that both follow UNIX
format. Only difference with a live system will be the password hash
field, which is stored in `passwd` instead of a dedicated `shadow` file
(and which is likely encoded differently).

### CODE REVIEW

File: src/passwd.cr [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+		CSV.parse(line[4], separator: ',')[0]?.try do |gecos|
+			@full_name = gecos[0]?
+			@location = gecos[1]?
+			@office_phone_number = gecos[2]?
+			@home_phone_number = gecos[3]?
+			@other_contact = gecos[4]?
+		end
+
+		# FIXME: What about those two fields? Keep them, remove them?
```
Comment (blankoworld) [inline src/passwd.cr]: For which kind of use? If not necessary now, don't use it.

File: src/passwd.cr [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+		users = get_all_users
+
+		while users.any? &.gid.== gid
+			gid += 1
+		end
+
+		gid
+	end
+
+	def add_user(login, password = nil, uid = nil, gid = nil, home = "/", shell = "/bin/nologin")
```
Comment (blankoworld) [inline src/passwd.cr]: Exception: "User already exists"

File: src/passwd.cr [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+	##
+	# Will fail if the user is found but the password is invalid.
+	def get_user(login : String, password : String) : AuthD::User?
+		digest = OpenSSL::Digest.new("sha256")
+		digest << password
+		hash = digest.hexdigest
+
+		each_user do |user|
+			if user.login == login
+				# FIXME: XXX: HASH!!!!!
```
Comment (blankoworld) [inline src/passwd.cr]: I don't understand what's to be fixed. Any idea?

**Comment by KaneRoot** [comment]
Everything seems good to me. I like the idea to remove dependencies to external tools like SGBDs. This still need some more functionalities, such as modifying and removing users, but this is a great start.

File: src/passwd.cr (hunk 3 above)
Comment (Lukc) [reply src/passwd.cr]: I think I fixed it already and forgot the FIXME. :smile:

File: src/passwd.cr (hunk 1 above)
Comment (Lukc) [reply src/passwd.cr]: Well… they’re not being used right now, but it’s necessary to import them if we want to be able to re-export or rewrite the passwd files. In the end, I guess that FIXME should also be removed. :man_shrugging:

**Pull request merged**